### DISCUSSION TITLE
TimeoutError on TBNB transfer using Wagmi and Viem in Next.js

### DISCUSSION THREAD

**Discussion by karangoraniya** [wevm/wagmi#2674] [Q&A] [answered]
**Description:**
I'm encountering a TimeoutError when I try to transfer TBNB using Wagmi and Viem in a Next.js application. The exact error message is:

**Expected Behavior:**
It should transfer the native token of blockchain to destination address.

**Actual Behavior:**

```bash
TimeoutError: The request took too long to respond.
URL: https://data-seed-prebsc-1-s1.binance.org:8545/
Request body: {"method":"eth_getBalance","params":["0x1f.....472B","latest"]}
Details: The request timed out.
Version: viem@1.1.7
```

```bash
EstimateGasExecutionError: The request took too long to respond.

URL: https://data-seed-prebsc-1-s1.binance.org:8545
Request body: {"method":"eth_estimateGas","params":[{"from":"0x1f2EC977469F1A036E1b231109C63638a212472B","gasPrice":"0x430e23400","to":"0xEC0C6a58e2339428d9DE08640F43D5b0b3ED0606","value":"0x0"}]}
 
Estimate Gas Arguments:
  from:      0x1f2....472B
  to:        0xEC....0606
  value:     0.1 tBNB
  gasPrice:  18 gwei

Details: The request timed out

```

**Note: This issue does not occur when I'm using other networks like Polygon Mumbai, Sepolia, Goerli etc.., only with TBNB.**

**Additional Information:**
```ts
 const { config } = usePrepareSendTransaction({
        to: to,
        value: parseEther(amount),
        gasPrice: BigInt(18000000000),
    })
```

**Environment:**
- OS: Mac
- Node.js version: v20.3.1
- Wagmi version: 1.2.2
- Viem version: 1.1.8
- Next.js version: Next 13 App Routing

**Comment by SaadNiaziDev** [comment, marked as answer]
try to override testnet default rpc. You can do that by in this way.
`
import {bscTestnet} from 'wagmi/chains';
`
`
bscTestnet.rpcUrls.default = 'ANY_RPC_FROM_CAHINLIST.ORG'
`
then pass this chain in wagmi configureChains, see if it works or not

**Reply by karangoraniya** [reply]
@SaadNiaziDev 
Thank you. I have added a custom chain, and your implementation is working fine. I just need to add an interface in TypeScript.
